sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/AggregateDocumentQueryExecutionContext.java

```diff
@@ -84,7 +84,7 @@ public Flux<FeedResponse<Document>> drainAsync(int maxPageSize) {
                         for(String key : BridgeInternal.queryMetricsFromFeedResponse(page).keySet()) {
                             if (queryMetricsMap.containsKey(key)) {
                                 QueryMetrics qm = BridgeInternal.queryMetricsFromFeedResponse(page).get(key);
-                                queryMetricsMap.get(key).add(qm);
+                                queryMetricsMap.put(key, QueryMetrics.addQueryMetrics(queryMetricsMap.get(key), qm));
```

Comment:
should we use compute()

Reply:
Good point, done, and some more optimizations.